vcd/datasource_vcd_catalog.go

```diff
@@ -139,5 +170,33 @@ func datasourceVcdCatalogRead(ctx context.Context, d *schema.ResourceData, meta
 		return diag.Errorf("There was an issue when setting metadata into the schema - %s", err)
 	}
 
+	numberOfVAppTemplates, err := catalog.QueryVappTemplateList()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to retrieve vApp templates associated to this catalog: %s", err)
+		return diag.Errorf("There was an issue when retrieving vApp templates - %s", err)
+	}
+
+	dSet(d, "number_of_vapp_templates", len(numberOfVAppTemplates))
+
+	numberOfMedia, err := catalog.QueryMediaList()
+	if err != nil {
+		log.Printf("[DEBUG] Unable to retrieve media items associated to this catalog: %s", err)
+		return diag.Errorf("There was an issue when retrieving media items - %s", err)
+	}
+
+	dSet(d, "number_of_media", len(numberOfMedia))
+
+	dSet(d, "is_published", catalog.Catalog.IsPublished)
+
+	controlAccessParams, err := catalog.GetAccessControl(false)
+	if err != nil {
+		return diag.Errorf("error retrieving catalog access control: %s", err)
+	}
+	if controlAccessParams.IsSharedToEveryone || controlAccessParams.AccessSettings != nil {
```

Comment:
Just double checking. Are we guaranteed the catalog is shared if `controlAccessParams.AccessSettings != nil`?

Reply:
Repeated comment